### PR TITLE
FCA: car port for Dodge Durango 2021

### PR DESCRIPTION
**Pull request by jnewb1**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/fingerprints.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: ```3379c85aeedc8285|2023-12-07--17-49-39```
- [ ] route with stock system:

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Version 0.9.6 (2024-02-XX)
 * Improved fuzzy fingerprinting for many makes and models
 * Hyundai Staria 2023 support thanks to sunnyhaibin!
 * Kia Niro Plug-in Hybrid 2022 support thanks to sunnyhaibin!
+* Dodge Duranago 2020-21 support
```
Comment (sshane) [inline RELEASES.md:14]: sort this

File: selfdrive/car/chrysler/values.py [hunk 2]
```diff
@@ -23,6 +23,8 @@ class CAR(StrEnum):
   PACIFICA_2018 = "CHRYSLER PACIFICA 2018"
   PACIFICA_2020 = "CHRYSLER PACIFICA 2020"
 
+  DURANGO = "DODGE DURANGO 2021"
```
Comment (sshane) [inline selfdrive/car/chrysler/values.py:26]:
```suggestion
  DODGE_DURANGO = "DODGE DURANGO 2021"
```

Any platform that isn't the current brand should generally be prefixed with it (LEXUS, JEEP, RAM, etc.)

File: selfdrive/car/tests/routes.py [hunk 3]
```diff
@@ -46,6 +46,7 @@ class CarTestRoute(NamedTuple):
   CarTestRoute("3d84727705fecd04|2021-05-25--08-38-56", CHRYSLER.PACIFICA_2020),
   CarTestRoute("221c253375af4ee9|2022-06-15--18-38-24", CHRYSLER.RAM_1500),
   CarTestRoute("8fb5eabf914632ae|2022-08-04--17-28-53", CHRYSLER.RAM_HD, segment=6),
+  CarTestRoute("3379c85aeedc8285|2023-12-07--17-49-39", CHRYSLER.DURANGO, segment=3),
```
Comment (sshane) [inline selfdrive/car/tests/routes.py:49]: why hard code the segment?

File: selfdrive/car/chrysler/fingerprints.py [hunk 4]
```diff
@@ -612,4 +612,34 @@
       b'M2421132MB',
     ],
   },
+  CAR.DURANGO: {
+    (Ecu.combinationMeter, 0x742, None): [
+      b'68454261AD',
+      b'68471535AE',
```
Comment (sshane) [inline selfdrive/car/chrysler/fingerprints.py:618]: can you post dongles for all FW?

**Review by sshane — CHANGES_REQUESTED** [review]

File: selfdrive/car/chrysler/fingerprints.py (hunk 4 above)
Comment (jnewb1) [reply selfdrive/car/chrysler/fingerprints.py:618]: 4bb5387dcb2bb6e8
3379c85aeedc8285

File: RELEASES.md (hunk 1 above)
Comment (jnewb1) [reply RELEASES.md]: done

File: selfdrive/car/chrysler/values.py (hunk 2 above)
Comment (jnewb1) [reply selfdrive/car/chrysler/values.py]: done

File: selfdrive/car/tests/routes.py (hunk 3 above)
Comment (jnewb1) [reply selfdrive/car/tests/routes.py]: removed

File: Jenkinsfile [hunk 5]
```diff
@@ -16,6 +16,8 @@ ssh -tt -o ConnectTimeout=30 -o ServerAliveInterval=30 -o ServerAliveCountMax=3
 
 set -e
 
+shopt -s huponexit # kill all child processes when the shell exits
+
```
Comment (sshane) [inline Jenkinsfile:20]:
```suggestion
```

Comment (jnewb1) [reply Jenkinsfile:20]: 1 sec merge failed

File: selfdrive/car/chrysler/values.py [hunk 6]
```diff
@@ -23,6 +23,8 @@ class CAR(StrEnum):
   PACIFICA_2018 = "CHRYSLER PACIFICA 2018"
   PACIFICA_2020 = "CHRYSLER PACIFICA 2020"
 
+  DODGE_DURANGO = "DODGE DURANGO 2021"
```
Comment (sshane) [inline selfdrive/car/chrysler/values.py:26]:
```suggestion
  # Dodge
  DODGE_DURANGO = "DODGE DURANGO 2021"
```

File: selfdrive/car/ford/carstate.py [hunk 7]
```diff
@@ -54,7 +54,7 @@ def update(self, cp, cp_cam):
     ret.steeringPressed = self.update_steering_pressed(abs(ret.steeringTorque) > CarControllerParams.STEER_DRIVER_ALLOWANCE, 5)
     ret.steerFaultTemporary = cp.vl["EPAS_INFO"]["EPAS_Failure"] == 1
     ret.steerFaultPermanent = cp.vl["EPAS_INFO"]["EPAS_Failure"] in (2, 3)
-    # ret.espDisabled = False  # TODO: find traction control signal
+    ret.espDisabled = cp.vl["Cluster_Info1_FD1"]["DrvSlipCtlMde_D_Rq"] != 0  # 0 is default mode
```
Comment (sshane) [inline selfdrive/car/ford/carstate.py:57]: looks like a merge conflict

Comment (jnewb1) [reply selfdrive/car/ford/carstate.py]: fixed

**Review by sshane — APPROVED** [review]

File: selfdrive/car/chrysler/fingerprints.py (hunk 4 above)
Comment (sshane) [reply selfdrive/car/chrysler/fingerprints.py:618]: connect shows 4bb5387dcb2bb6e8's routes as green when below 40 mph, did you check if any of these can steer to zero?

Comment (sshane) [reply selfdrive/car/chrysler/fingerprints.py:618]: State 8 supposedly means you can steer, interesting, looks like you can steer to zero once you drive above 38 kph for the first time

image

Comment (sshane) [reply selfdrive/car/chrysler/fingerprints.py:618]: `3379c85aeedc8285` only allows steering at 40 mph, and then stops steering at 35 mph though, so the higher limit for this PR is fine, we can detect this later as an enhancement

**Pull request merged**